src/xdocs/beginning_development.xml

```diff
@@ -31,15 +31,21 @@
         <a href="https://help.github.com/articles/fork-a-repo/"> here</a><br/>
         3. Checkout the current source code from: https://github.com/you_user_name/checkstyle/<br/>
         by running
-        <source class="wrap-content">
-git clone git@github.com:you_user_name/checkstyle.git
+      </p>
+      <div class="wrap-content">
+        <source>
+            git clone git@github.com:you_user_name/checkstyle.git
```

Comment:
no indentation is intended.
This is the same for all similar to this. This is to help prevent hitting the 100 character line limit.

Reply:
Well, with my changes you can indent it or not, it doesn't make any difference in the final result.
I can of course unindent the parts again, but don't you think it looks cleaner indented?

Reply:
It may look neater in the source code, but it also carried through to the web page. And we have add issues with line lengths as stated before.
I think we would like to keep them unindented.

Reply:
I don't get what you mean.
The `MissingOverride` example is not marked with `class` `wrap-content` which is why it is strangely indented.
If it were marked with `class` `wrap-content` and my changes applied, the leading whitespace is stripped and so it would look proper, no matter whether you indent it in the source or not.

Again, with my changes the `.wrap-content` class changed from `white-space: pre-wrap` to `white-space: pre-line` where the leading whitespace is stripped in the display, so it makes absolutely no difference in the end-result whether you indent the example in the source or not.

Reply:
> wrap-content which is why it is strangely indented.

@Vampire I didn't know thats what the `wrap-content` changes were for.
How will it handle consecutive lines? If it removes all indentation then that will cause its own issue.

Instead of:
````
<module name="MissingOverride">
  <property name="javaFiveCompatibility"
    value="true"/>
</module>
````

We will  get:
````
<module name="MissingOverride">
<property name="javaFiveCompatibility"
value="true"/>
</module>
````

Which is not easy to read.
It may also make the display weird if we have a long line that we need to break apart more as it gets deeply nested. It looks like `wrap-content` still keeps the line breaks, so deeply nested sources will have very short lines in source and display.
Unless you can provide another alternative, then I think what we had before works better for us.

Reply:
we did no proper indentation to keep our source code render in HTML with no indentation.
I do not like our approach also, but I do not know for now how to make it better. If you know please share an approach and lets move this to separate commit/PR, or it is caused by plugin validation?.

Reply:
The `.source` style for the `<source>` tags has `white-space: nowrap;` which means no automatic wrapping is done by the browser.
This then caused #3902.
To fix this the `wrap-content` class was introduced which set `white-space: pre-wrap !important;` for selected cases.
This caused the whitespaces to be preserved but if necessary line-wrapping to be done and thus also required all relevant code parts to be unindented.
Additionally as it preserves linebreaks, it looks like this (extra line in the end):
![grafik](https://user-images.githubusercontent.com/325196/53968521-85f47100-40f7-11e9-8264-1375d823ea5d.png)
I wanted to look nicer, like this:
![grafik](https://user-images.githubusercontent.com/325196/53969156-8b05f000-40f8-11e9-951d-752a89884827.png)
Due to that part of my changes is to use `white-space: pre-line !important;` instead for the `pre` tag. As it strips the whitespace, the strange extra line is gone without the need to always have the `</source>` tag on the last line of the actual content which would be an alternative. A side-effect of this is, that the code parts can be indented properly in the source, as it causes the whitespaces to be stripped, but newlines to be preserved and still wrap if necessary.

For the intentional indenting that you stripped in your "We will get" example are the non-breaking spaces.
Also, this is only for the parts marked with the `wrap-content` class.

Reply:
I wasn't aware the changes to the styles removed the extra spaces in front.
I also see now this wasn't done for all sources, just the wrapped content ones.

I did however like the no indentation for these items. It made seeing the internal flow easier as it stood out from the actual xml. It also allowed wraps and non-wraps to have the same indentation make up so you wouldn't accidentally add the indentation to one when it shouldn't have. I also can't say I'm a fan of the `&#xa0;` just because it isn't a common thing I've seen.

I will leave this up to @romani to decide on. Issue still isn't approved.